packages/serverpod/lib/src/database/value_encoder.dart

```diff
@@ -20,6 +20,22 @@ class ValueEncoder extends PostgresTextEncoder {
           escapeStrings: escapeStrings);
     } else if (input is UuidValue) {
       return "'${input.uuid}'";
+    } else if (input is Enum) {
+      // Determine whether the Enum is serialized as an integer or as a String
+      var enumValueAsJson = (input as dynamic)?.toJson();
+      if (enumValueAsJson is int) {
```

Comment:
Question: For my knowledge - What happens in the case that the enum string is a series of numbers. For example if we have a string that is `"12345"`. Would this parse it correctly? Is that the reson you use the `toJson()` over `int.parse(...)`?

Reply:
(It might not be a real case since enum values can't be numbers, but I'm just curious.)

Reply:
I copied that code snippet `(input as dynamic)?.toJson()` from somewhere else in Serverpod when I first wrote this code!

This is converting an int or string to a string though. `int.parse` would go the other direction.